dali/kernels/imgproc/geom/remap.h

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_GEOM_REMAP_H_
+#define DALI_KERNELS_IMGPROC_GEOM_REMAP_H_
+
+#include <optional>
+#include "dali/core/common.h"
+#include "dali/kernels/kernel.h"
+#include "include/dali/core/geom/box.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * Specifies, how to handle pixels on a border of an image.
+ *
+ * --------------------------------------------------------------|
+ * | CONSTANT    | iiiiii|abcdefgh|iiiiiii with some specified i |
+ * --------------------------------------------------------------|
+ * | REPLICATE   | aaaaaa|abcdefgh|hhhhhhh                       |
+ * --------------------------------------------------------------|
+ * | REFLECT     | fedcba|abcdefgh|hgfedcb                       |
+ * --------------------------------------------------------------|
+ * | REFLECT_101 | gfedcb|abcdefgh|gfedcba                       |
+ * --------------------------------------------------------------|
+ * | WRAP        | cdefgh|abcdefgh|abcdefg                       |
+ * --------------------------------------------------------------|
+ * | TRANSPARENT | uvwxyz|abcdefgh|ijklmno                       |
+ * --------------------------------------------------------------|
+ * | ISOLATED    | do not look outside of ROI                    |
+ * --------------------------------------------------------------|
+ */
+enum class BorderType {
+  CONSTANT,
+  REPLICATE,
+  REFLECT,
+  REFLECT_101,
+  WRAP,
+  TRANSPARENT,
+  ISOLATED
+};
+
+template <typename T>
+struct Border {
+  BorderType type;
+  std::optional<T> value;
```

Comment:
Optional won't work if we include this in any .cu file.

Reply:
Changed to `T value`.